### PR TITLE
Support fireman carrying people onto morgue trays and other bucklable objects (stasis beds, operating tables, chairs)

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request

This PR lets you click on open trays while fireman carrying on them to place them on it.

## Why It's Good For The Game

Currently, if you are fireman carrying someone, the best course of action is to close the tray if it is open, stand on top of the tile, drop the person, stand out of the way, then close the tray. You might be able to drag and drop their sprite as well, though this is still not as intuitive as simply clicking on the tray.

## Changelog
:cl:
add: You can now place people you are fireman carrying onto morgue trays.
add: You can now place people you are fireman carrying onto objects that buckle players.
/:cl:

### CODE REVIEW

**Comment by Twaticus** [comment]
wow i thought you meant food trays until i read the branch name wtf

**Comment by Jared-Fogle** [comment]
Changed it to be more obvious then.

**Comment by WarlockD** [comment]
*obligatory vore refrence*  Just curious, why do you call them trays?  Not that I have a better name mind you.

**Comment by XDTM** [comment]
If i'm not wrong, it's because actual morgues keep bodies on large trays, i suppose to make it easier to move them around. In SS13 they behave more like drawers, but i suppose the original term stuck for realism.

**Comment by Jared-Fogle** [comment]
*I* called them trays just because that's what it's code named.

**Comment by Iatots** [comment]
Perhaps you should be able to set people down on roller beds too?

**Comment by Jared-Fogle** [comment]
@Iatots I can do that.

**Comment by Jared-Fogle** [comment]
Before I commit that, can anyone think of a scenario of an oversight with that? Fireman carrying someone, clicking on something that can buckle, and it putting them in like normal?

**Comment by Twaticus** [comment]
im still laughing about carrying people on food trays

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
please format both to be

```DM
if(!check)
    return ..()
rest of code
```

Also if it meets the req it will not send the attackbysignal

**Comment by Jared-Fogle** [comment]
@ExcessiveUseOfCobblestone Will do that tonight.

**Comment by Jared-Fogle** [comment]
I have since my message made those changes.

**Comment by Jared-Fogle** [comment]
It's been ready for a long time.

File: code/game/objects/buckling.dm [hunk 1]
```diff
@@ -20,6 +20,17 @@
 			if(user_unbuckle_mob(buckled_mobs[1],user))
 				return 1
 
+/atom/movable/attackby(obj/item/W, mob/user, params)
+	if(!can_buckle || !istype(W, /obj/item/riding_offhand))
+		return ..()
```
Comment (Jared-Fogle) [inline code/game/objects/buckling.dm:25]: Looking at this condition again now--what happens if you attack a chair? Will check to make sure it's normal.

Comment (Jared-Fogle) [reply code/game/objects/buckling.dm:25]: It works fine, but I don't know if that's just a lucky coincidence based on the ordering.

**Pull request merged**